s2wav/Dimension helper functions.py

```diff
@@ -0,0 +1,370 @@
+    
+    Note: Computes the number of samples in [0, pi], *not* over extended domain.
+
+    Args:
+        L: Harmonic band-limit.
+    
+    Returns:
+        ntheta: Number of theta samples.   
+    '''
+    return L+1
+
+
+
+def ssht_sampling_mw_ntheta(L: int) -> int:
+    '''Computes the number of theta samples for McEwen and Wiaux sampling.
+
+    Note: Computes the number of samples in (0, pi], *not* over extended domain.
+
+    Args:
+        L: Harmonic band-limit.
+    
+    Returns:
+        ntheta: Number of theta samples.
+    '''
+    return L
+
+
+
+
+#Functions from lines 41 - 110 of s2let_tiling.c
+
+
+
+def s2let_bandlimit(s2let_kernel: str, lam: float, j: int) -> int:
+    '''Computes the band-limit of a specific wavelet scale.
+
+    Args:
+        s2let_kernel: The wavelet type.
+        lam: Wavelet parameter which determines the scale factor between consecutive wavelet scales.
+        j: Wavelet scale.
+    
+    Returns:
+        Band-limit.
+    '''
+    if s2let_kernel == "S2DW":
+        return math.ceil(lam ** (j+1))
+    else:
+        #This code currently does not support other wavelet types.
+        return print("Invalid wavelet type.")
+
+
+
+def s2let_L0(s2let_kernel: str, lam: float, j: int) -> int:
+    '''Computes the minimum harmonic index supported by the given wavelet scale.
+
+    Args:
+        s2let_kernel: The wavelet type.
+        lam: Wavelet parameter which determines the scale factor between consecutive wavelet scales.
+        j: Wavelet scale.
+    
+    Returns:
+        el_min.
+    '''
+    if s2let_kernel == "S2DW":
+        return math.ceil(lam ** (j-1))
+    else:
+        #This code currently does not support other wavelet types
+        return print("Invalid wavelet type.")
+
+
+
+def s2let_j_max(L: int, lam: float) -> int:
+    '''Computes needlet maximum level required to ensure exact reconstruction.
+
+    Args:
+        L: Upper harmonic band-limit.
+        lam: Wavelet parameter which determines the scale factor between consecutive wavelet scales.
+    
+    Returns:
+        j_max
+    '''
+    return math.ceil(log(L) / log(lam))
+
+
+
+
+
+#files originally from s2let_helper
+
+def s2let_n_phi(sampling_scheme: str, L: int) -> int:
+    '''Returns the number of phi samples.
+
+    Args:
+        sampling_scheme: Either S2LET_SAMPLING_MW or S2LET_SAMPLING_MW_SS.
+        L: Upper harmonic band-limit.
+
+    Returns:
+        nphi: Number of phi samples.
+    '''
+    if sampling_scheme == "S2LET_SAMPLING_MW_SS":
+        return ssht_sampling_mw_ss_nphi(L)
+    else:
+        return ssht_sampling_mw_nphi(L)
+
+
+def s2let_n_theta(sampling_scheme: str, L: int) -> int:
+    '''Returns the number of theta samples.
+
+    Args:
+        sampling_scheme: Either S2LET_SAMPLING_MW or S2LET_SAMPLING_MW_SS.
+        L: Upper harmonic band-limit.
+
+    Returns: 
+        ntheta: Number of theta samples.
+    '''
+    if sampling_scheme == "S2LET_SAMPLING_MW_SS":
```

Comment:
When comparing against a string we can be a bit more careful. For instance if a user chooses "MW" it will not parse this to be "mw" automatically. 

 We can account for this by forcing the string to always be lower-case. Whenever we have if statements like this we should do something like 

`if sampling_scheme.lower() == "mw":`

where .lower() just takes a string and converts it to lower case (if it is not already).